packages/collector/src/announceCycle/agentHostLookup.js

```diff
@@ -127,37 +120,67 @@ function checkHost(host, cb) {
         port: agentOpts.port,
         path: '/',
         agent: http.agent,
-        method: 'GET'
+        method: 'GET',
+        // timeout for establishing a connection
+        timeout: requestTimeout
```

Comment:
When testing various error conditions I noticed that the timeout of 5 seconds that we previously set like this:

```
 req.setTimeout(5000, function onTimeout() {
    cb(new Error('Host check timed out'));
  });
```
 
did not have any effect. Then I remembered that the behavior around that changed in Node.js 9 -- and the original code was definitely written before Node.js 9.

Nowadays, `req.setTimeout(5000, ...)` will only set a timeout _after_ the connection has been established (that is, it is effective if the server takes to long to send a response); but it will not create a timeout if we cannot even connect to the host/port. 

Doing both (adding the timeout option and also calling req.setTimeout) handles both cases.

Reply:
Good catch